pkg/worker/worker.go

```diff
@@ -0,0 +1,185 @@
+package worker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	logger "log"
+	"os/exec"
+	"sync"
+	"syscall"
+
+	"github.com/google/uuid"
+	"github.com/renatoaguimaraes/job-scheduler/pkg/worker/conf"
+	"github.com/renatoaguimaraes/job-scheduler/pkg/worker/log"
+)
+
+// Command is a job request with the program name and arguments.
+type Command struct {
+	// Name program path/name
+	Name string
+	// Args program arguments
+	Args []string
+}
+
+// Job represents an arbitrary Linux process schedule by the Worker.
+type Job struct {
+	// ID job identifier
+	ID string
+	// Command pipeline
+	Cmd *exec.Cmd
+	// Status of the process.
+	Status *Status
+}
+
+// IsRunning checks if the process still running.
+func (j *Job) IsRunning() bool {
+	return j.Status.ExitCode != 0 || !j.Status.Exited
```

Comment:
why does a non-zero status code mean that the process is still running?

Reply:
You are right. I will change to `j.Status.ExitCode == 0 && !j.Status.Exited`